controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -292,23 +307,75 @@ func (r *Reconciler) hostOperatorHandleStatus(reqLogger logr.Logger, toolchainSt
 
 	// check host operator deployment status
 	deploymentConditions := status.GetDeploymentStatusConditions(r.Client, hostOperatorDeploymentName, toolchainStatus.Namespace)
-	err = status.ValidateComponentConditionReady(deploymentConditions...)
-	if err != nil {
-		reqLogger.Error(err, "host operator deployment is not ready")
+	errDeploy = status.ValidateComponentConditionReady(deploymentConditions...)
+	if errDeploy != nil {
+		operatorStatus.Conditions = deploymentConditions
+		toolchainStatus.Status.HostOperator = operatorStatus
+		reqLogger.Error(errDeploy, "host operator deployment is not ready")
+		return false
 	}
-	// update toolchainStatus
 	operatorStatus.Conditions = deploymentConditions
+
+	// if we are running in production we also
+	// check that deployed version matches the latest commit from source code repository.
+	// we also need to check when we called GitHub api last time, in order to avoid rate limiting issues.
+	toolchainConfig, errToolchainConfig := toolchainconfig.GetToolchainConfig(r.Client)
+	if errToolchainConfig != nil {
+		reqLogger.Error(errToolchainConfig, status.ErrMsgCannotGetDeployment)
+		errCondition := status.NewDeploymentVersionUpToDateErrorCondition(toolchainv1alpha1.ToolchainStatusDeploymentVersionCheckOperatorErrorReason,
+			fmt.Sprintf("unable to get ToolchainConfig: %s", errToolchainConfig.Error()), corev1.ConditionUnknown)
+		operatorStatus.Conditions = append(operatorStatus.Conditions, *errCondition)
+		toolchainStatus.Status.HostOperator = operatorStatus
+		return false
+	}
+	if !isProdEnvironment(toolchainConfig) {
+		// version check is disabled
+		deploymentVersionSkippedCondition := status.NewDeploymentVersionUpToDateCondition(toolchainv1alpha1.ToolchainStatusDeploymentVersionCheckDisabledReason)
+		operatorStatus.Conditions = append(operatorStatus.Conditions, *deploymentVersionSkippedCondition)
+		toolchainStatus.Status.HostOperator = operatorStatus
+		return true
+	}
+
+	if !client.CanIssueGitHubRequest(r.lastGitHubAPICallForHost) {
+		// we need to wait until delay is over before calling again GitHub API
+		toolchainStatus.Status.HostOperator = operatorStatus
+		return true
+	}
```

Comment:
from the comment in toolchain-common repo https://github.com/codeready-toolchain/toolchain-common/pull/315#discussion_r1230818952
```
// we need to make the gap between the calls for this repo, but we have to set the condition to something - let's reuse what was set last time
```
I was referring to this part - you don't set/reuse the original condition, which means that the previously condition will be gone after 5s when the controller reconciles again